File: src/twoSum.cpp
#include <vector>
#include <map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        map<int, int> map;
        vector<int> result;

        for (int i = 0; i < nums.size(); i++)
        {
            int number = nums[i];
            int desiredNum = target - number;
            if (map.find(number) == map.end())
            {
                map[desiredNum] = i;
            }
            else
            {
                result.push_back(map[number]);
                result.push_back(i);
                return result;
            }
            
        }
        
        return result;
    }
};

File: src/LeetCodePY/twoSum.py
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        dict = {}

        for i in range(len(nums)):
            if nums[i] in dict:
                return [dict[nums[i]], i]

            dict[target - nums[i]] = i
        return []
